src/esp32/src/flash.c

```diff
@@ -6,14 +6,147 @@
 
 #include <target/flash.h>
 
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <log.h>
+#include <private/rom.h>
+#include <private/soc_utils.h>
+
+#include "soc/reg_base.h"
+#include "soc/spi_reg.h"
+
+#define SPI_FLASH_CMD_RDID 0x9F
+#define ESP_ROM_SPIFLASH_BUSY_FLAG BIT(0)
+
+#define GPIO_STRAP_REG (DR_REG_GPIO_BASE + 0x0038)
+#define PERIPHS_SPI_FLASH_CMD SPI_CMD_REG(1)
+#define PERIPHS_SPI_FLASH_STATUS SPI_RD_STATUS_REG(1)
+#define PERIPHS_SPI_FLASH_USRREG SPI_USER_REG(1)
+#define PERIPHS_SPI_FLASH_USRREG2 SPI_USER2_REG(1)
+#define PERIPHS_SPI_FLASH_C0 SPI_W0_REG(1)
+#define PERIPHS_SPI_MOSI_DLEN_REG SPI_MOSI_DLEN_REG(1)
+#define PERIPHS_SPI_MISO_DLEN_REG SPI_MISO_DLEN_REG(1)
+
+/* ROM */
+
+extern uint32_t ets_efuse_get_spiconfig(void);
+extern void esp_rom_spiflash_attach(uint32_t ishspi, bool legacy);
+extern esp_rom_spiflash_chip_t g_rom_flashchip;
+extern uint8_t g_rom_spiflash_dummy_len_plus[];
```

Comment:
These definitions are used for all chips. So seems to be reasonable t move them to one header file like it is done in IDF. 
- https://github.com/espressif/esp-idf/blob/master/components/esp_rom/include/esp_rom_spiflash.h
- https://github.com/espressif/esp-idf/blob/master/components/esp_rom/include/esp_rom_spiflash_defs.h

Reply:
Make the common code a bit different way, ptal